quality-assurance/core.md

```diff
@@ -0,0 +1,192 @@
+* Errors are handled correctly in try catch blocks
+* Responses are returned in a consistent &amp; logical format
+* Only appropriate &amp; relevant information is returned
+* When returning JSON ensure the returned JSON conforms to the [JSON spec](http://json-schema.org/specification.html)
+* Appropriate status codes are set
+
+**Emails**
+* Emails use the SES driver unless agreed beforehand
+* All emails are queued up for sending
+* Only relevant information is passed to the mailable instance
+
+**Views**
+* Loading of assets (JS/HTML) use the `asset` global helper
+* Blade layout system is used where appropriate (@extends, @yield)
+* Components are split out logically where appropriate, make use of Laravel's powerful blade templating system if 
+appropriate (clientCard, navbar etc)
+
+**Service Providers**
+* Service Providers are utilised where appropriate to bind classes into the container (API interfaces etc)
+
+**File Storage**
+* Files are stored on S3 unless requested by the client to be stored locally or agreed beforehand
+* Ensure appropriate permissions on files in applicable (sensitive documents are private, avatars are public etc)
+
+**Queues**
+* Queues use Redis unless agreed beforehand
+* Ensure appropriate measures are in place for handling failed jobs
+
+**Broadcasting**
+* Broadcasting uses Laravel Echo unless agreed beforehand
+* Ensure appropriate authentication & security for each channel
+
+**Database**
+* All migrations use foreign keys for relations to ensure database integrity
+* SQL data types are appropriate
+* Monetary values should either be stored in pence as an INTEGER or as a DECIMAL - Never a FLOAT
+* Large tables that are queried often are indexed in order to keep database speed up
+
+## Processes
+### Automated Testing
+
+Automated testing should be implemented in a way to confirm functionality specified by the client.
+To this end, automated tests should be created to confirm the project user stories and related 
+business logic is satisfied and correct. 
+
+#### Unit Testing
+
+Our current policy is that unit tests should be used for:
+
+* Testing of project business logic / rules, that are able to be sufficient abstracted
+* Testing of packages, including our open source work
+
+#### Integration, Browser &amp; End-to-End Testing
+
+Integration, browser and end-to-end tests should be used for:
+
+* Testing of project functionality related to client specified user stories
+* Testing of critical path functionality, such as the register, login, and checkout process of an
+ecommerce site
+
+Since end-to-end tests are slower than browser tests and browser tests are slower than integration 
+tests, you should ideally aim to use the fastest possible test type, if appropriate.
+### Manual Testing
+#### 1 - Developer Testing
+During the development of the project, developers will create `feature/*` branches for all features going into the software.
```

Comment:
I don't think this statement is needed since we already have this highlighted in our git flow documentation.

Reply:
@ash123456789 Agreed. 👍 

```suggestion
During the development of the project, developers will create branches as described in our git work flow documentation.
```